packages/framework/src/Command/FixExtendedClassesAnnotationsCommand.php

```diff
@@ -0,0 +1,123 @@
+     */
+    protected $projectRootDirectory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry
+     */
+    protected $classExtensionRegistry;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Replaces the framework with the project annotations in all project files when there exists a project extension of a given framework class.');
+    }
+
+    /**
+     * @param string $projectRootDirectory
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry $classExtensionRegistry
+     */
+    public function __construct(string $projectRootDirectory, ClassExtensionRegistry $classExtensionRegistry)
+    {
+        $this->projectRootDirectory = $projectRootDirectory;
+        $this->classExtensionRegistry = $classExtensionRegistry;
+        parent::__construct();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->replaceFrameworkWithProjectAnnotations();
+        $output->writeln('Annotations fixed successfully');
+
+        return 0;
+    }
+
+    protected function replaceFrameworkWithProjectAnnotations(): void
+    {
+        $finder = $this->getFinderForReplacingAnnotations();
+        $annotationsReplacementsMap = $this->getAnnotationsReplacementsMap($this->classExtensionRegistry->getServiceExtensionMap());
+        foreach ($finder as $fileWithFrameworkAnnotationsToReplace) {
+            $pathname = $fileWithFrameworkAnnotationsToReplace->getPathname();
+            $content = file_get_contents($pathname);
+            $replacedContent = preg_replace(
+                array_keys($annotationsReplacementsMap),
+                $annotationsReplacementsMap,
+                $content
+            );
+            file_put_contents($pathname, $replacedContent);
+        }
+    }
+
+    /**
+     * @return \Symfony\Component\Finder\Finder
+     */
+    protected function getFinderForReplacingAnnotations(): Finder
+    {
+        $finder = Finder::create()
+            ->files()
+            ->ignoreUnreadableDirs()
+            ->in([
+                $this->projectRootDirectory . '/src',
+                $this->projectRootDirectory . '/tests',
+            ])
+            ->name('*.php')
+            ->contains($this->getRelevantFrameworkAnnotationsPattern());
+
+        return $finder;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getRelevantFrameworkAnnotationsPattern(): string
+    {
+        $frameworkClasses = array_keys($this->classExtensionRegistry->getServiceExtensionMap());
+        $escapedFrameworkClasses = array_map(function ($frameworkClass) {
+            return str_replace('\\', '\\\\', $frameworkClass);
+        }, $frameworkClasses);
+
+        return sprintf('/(@var|@param|@return) (\\\\%s)/', implode('|\\\\', $escapedFrameworkClasses));
```

Comment:
Wow 😄 that's a lot of backslashes.

What about to create the regular first and replace backslashes after? In that case, you could omit the whole array_map and it will be (hopefully) more readable

Reply:
I like your suggestion, applied in 921212897